Converted statics to be const

Dr15Jones opened the pull request.

Dr15Jones committed:
Converted statics to be const

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

FWCore/Framework


@cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @wddgit this is something you requested to watch as well.
@muzaffar, @davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @muzaffar, @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: 1929855ba8afd23cb5fdfb9790eca0306c04687e


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/0ca3ab24cecacf9a285d60d524b0d3a353502351
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19224/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19224/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19224/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/ValidityInterval.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/EDConsumerBase.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/EventSetupRecord.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/EventSetup.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/EDFilter.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/ESProducts.cc:20:42: error: default initialization of an object of const type 'const eventsetup::produce::Produce' without a user-provided default constructor
      const eventsetup::produce::Produce produced;
                                         ^
                                                 {}
1 error generated.
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-04-18-1100/src/FWCore/Framework/src/DataKeyTags.cc 
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/0ca3ab24cecacf9a285d60d524b0d3a353502351
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19224/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19224/git-merge-result

cmsbuild commented:
Comparison job queued.

Dr15Jones committed:
Added default constructor required by clang

clang requires that classes used as const globals must have an explicit default constructor.

cmsbuild commented:
Pull request #18391 was updated. @cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: bae1b2b70e25430ce20c43f4d984e75475d128ba
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19229/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/0ca3ab24cecacf9a285d60d524b0d3a353502351
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19229/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19229/git-merge-result

cmsbuild commented:
Comparison job queued.

Dr15Jones commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @muzaffar, @davidlange6, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18391/19229/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 1615 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1826273
 - DQMHistoTests: Total failures: 22644
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1803456
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files
